browser/src/control/Control.ContextToolbar.ts

```diff
@@ -0,0 +1,231 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * L.Control.NotebookbarWriter - definition of notebookbar content in Writer
+ */
+
+/* global _ _UNO app Point */
+
+class ContextToolbar {
+	container: HTMLElement;
+	builder: any;
+	map: any;
+	initialized: boolean = false;
+
+	constructor(map: any) {
+		this.map = map;
+		this.builder = new L.control.notebookbarBuilder({
+			windowId: -2,
+			mobileWizard: this,
+			map: this.map,
+			cssClass: 'notebookbar',
+			useSetTabs: true,
+		});
+		this.container = L.DomUtil.createWithId(
+			'div',
+			'context-toolbar',
+			document.body,
+		);
+		L.DomUtil.addClass(this.container, 'notebookbar horizontal');
+	}
+
+	showContextToolbar(pos: Point): void {
+		if (!this.initialized) {
+			this.builder.build(this.container, this.getWriterTextContext(), false);
+			this.initialized = true;
+		}
+
+		this.builder.map.on('jsdialogaction', this.onJSAction, this);
+		this.builder.map.on('jsdialogupdate', this.onJSUpdate, this);
+		app.layoutingService.appendLayoutingTask(() => {
+			this.showHideToolbarImp('show', pos);
+		});
+	}
+
+	hideContextToolbar(): void {
+		this.builder.map.off('jsdialogaction', this.onJSAction, this);
+		this.builder.map.off('jsdialogupdate', this.onJSUpdate, this);
+		app.layoutingService.appendLayoutingTask(() => {
```

Comment:
`app.layoutingService.appendLayoutingTask` could be done inside `showHideToolbarImpl` so if someone else will use that method - will out of the box do it right